### PR TITLE
(fix|COS-529): Fix open issue in preview modal from issue card behaviour + refactored context to limit rerenders

### PR DESCRIPTION
**Pull request by Kasia9090**
## Changes

- Refactored timeline preview modal context to better accommodate need to manage it from different places.
- Improved performance thanks to limiting amount of rerenders related to modal context usage. 
- Refactor timeline event components for cleaner context usage
- The new changes improve also readability by reorganizing the context into separate concern-specific hooks e.g useDeepLinkToOpenModal, useTimelineEventCachedData.
- The separation of state and methods into the context has been created for better performance. 
- These improvements make the code easier to analyze and modify.

Additionally, code related to timeline event context hook in components OrganizationTimeline and IssuePreviewModal has been refactored to reflect these changes. The updates also include minor changes to several files related to organization project.

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

### Summary by CodeRabbit

- Refactor: Improved the structure and readability of the codebase, particularly within the Timeline component. This will enhance maintainability and future development.
- New Feature: Introduced a new context provider to manage timeline events, improving the efficiency of data handling.
- Refactor: Updated the way modals are handled, making them more consistent and easier to manage.
- New Feature: Added deep linking functionality to open modals, enhancing user navigation and accessibility.
- Refactor: Updated the GraphQL client to accept additional configuration options, providing more flexibility for data fetching.
- Minor Text Changes: Updated the tooltip label in the PriorityBadge component for better user understanding.
  
Please note that these changes are mostly under-the-hood improvements and may not result in noticeable changes for end-users. However, they set the stage for more efficient and robust feature development in the future.

### CODE REVIEW

File: packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useTimelineEventCachedData.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const singleEventQueryKey = useGetTimelineEventsQuery.getKey({
+      ids: [timelineEventId],
+    });
+
+    const timelineQueryKey = useInfiniteGetTimelineQuery.getKey({
+      ...timelineMeta.getTimelineVariables,
+    });
+
+    return [singleEventQueryKey, timelineQueryKey];
+  };
```
[the remark this answers is not in the document]
Comment (Kasia9090) [reply packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useTimelineEventCachedData.ts:24]: this is a method that is not used as hook

File: packages/apps/spaces/app/organization/[id]/src/components/Timeline/events/logEntry/LogEntryStub.tsx [hunk 2]
```diff
@@ -34,7 +34,8 @@ function getAuthor(user: User | Contact) {
 }
 
 export const LogEntryStub = ({ data }: LogEntryStubProps) => {
-  const { openModal } = useTimelineEventPreviewContext();
+  const { openModal } = useTimelineEventPreviewMethodsContext();
+
   const fullName = getAuthor(data?.logEntryCreatedBy);
```
[the remark this answers is not in the document]
Comment (Kasia9090) [reply packages/apps/spaces/app/organization/[id]/src/components/Timeline/events/logEntry/LogEntryStub.tsx]: this comment is invalid. On click method in logEntryStub is correct

File: packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useDeeplinkToOpenModal.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            setModalContent(content);
+            setIsModalOpen(true);
+          }
+        });
+        return;
+      }
+      setModalContent(selectedEvent);
+      setIsModalOpen(true);
+    }
+  }, [searchParams]);
```
[the remark this answers is not in the document]
Comment (Kasia9090) [reply packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useDeeplinkToOpenModal.ts]: this is expected, this effect handles corder case in which we are interested only in search params change and content of modal content only in the very moment of that change

File: packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useDeeplinkToOpenModal.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          }
+        });
+        return;
+      }
+      setModalContent(selectedEvent);
+      setIsModalOpen(true);
+    }
+
+    return () => {
+      abortController.abort('Page unmounted');
```
[the remark this answers is not in the document]
Comment (Kasia9090) [reply packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/useDeeplinkToOpenModal.ts]:
```
interface AbortController {
    /**
     * Returns the AbortSignal object associated with this object.
     *
     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortController/signal)
     */
    readonly signal: AbortSignal;
    /**
     * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
     *
     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortController/abort)
     */
    abort(reason?: any): void;
}

```

from what I see in TS it accepts 'reason' as argument

**Review by xvasi — APPROVED** [review]

**Pull request merged**